Add Breadcrumb Heading Component

itisAliRH opened the pull request:
This pull request adds a `BreadcrumbHeading` component, enhancing user navigation by displaying the current page's hierarchical path. It replaces headers in `StorageManager`, `StorageDashboard`, and `Visualizations/OverviewPage` with this new component. Future PRs will extend its use to additional components.

|Before|After|
|--------|-------|
|![image](https://github.com/user-attachments/assets/b131f005-3b44-4ddc-b028-db6987c5c404)|![image](https://github.com/user-attachments/assets/3cbb96c0-eb88-430a-9b61-90b4d4d5e232)|
|![image](https://github.com/user-attachments/assets/26d49934-94a1-471e-b671-0cb31b06498c)|![image](https://github.com/user-attachments/assets/80f1b186-23c6-4b36-9e6c-aad9d9571f37)|
|![image](https://github.com/user-attachments/assets/c96d826c-6e29-4386-9e42-c39e1db530df)|![image](https://github.com/user-attachments/assets/b32fdd51-da79-4e90-9231-8c2a80a1adeb)|
|![image](https://github.com/user-attachments/assets/e7b708a1-4fd9-47d8-b246-c24346ff7837)|![image](https://github.com/user-attachments/assets/b386da3d-f61f-4fe4-865b-84bcc8c48d53)|

Key Changes:
- Addition of the `BreadcrumbHeading` component to display breadcrumb-style headings.
- Included a default slot within the component for header action buttons
![image](https://github.com/user-attachments/assets/4cea01a1-7bdb-42a7-a986-69ea840be8a5)
- Replace existing headings in `StorageManager`, `StorageDashboard`, and `Visualizations/OverviewPage` for consistent and improved user experience.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  1. Verify that the BreadcrumbHeading displays correctly and functions as expected in the updated components.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

itisAliRH committed:
Adds BreadcrumbHeading component

Introduces BreadcrumbHeading component for displaying breadcrumb navigation.
Utilizes Heading and RouterLink for structured and localized breadcrumb items.
Includes scoped styles for component layout.

itisAliRH committed:
Adds breadcrumb heading to WorkflowList

Replaces existing heading with BreadcrumbHeading for consistency in WorkflowList

itisAliRH committed:
Adds breadcrumb navigation to StorageDashboard

Replaces main header with Heading component for consistency
Enhances user experience by making navigation clearer

itisAliRH committed:
Adds breadcrumb navigation to StorageManager

Integrates BreadcrumbHeading component to replace the back link and header
Provides breadcrumb navigation items for better user experience
Removes scoped CSS for beta text styling

itisAliRH committed:
Add breadcrumb navigation to OverviewPage

Updates the UI to use BreadcrumbHeading for better navigation to enhance user experience

davelopez commented on client/src/components/User/DiskUsage/Management/StorageManager.vue:
```diff
@@ -11,11 +11,18 @@ import type { CleanableItem, CleanupOperation, CleanupResult } from "./Cleanup/m
 }
 
+const breadcrumbItems = [
+    { title: "User Preferences", to: "/user" },
```
Even if one of the ways to access the Storage Dashboard is through the user preferences, it is not really tied to the User preferences. So I would suggest removing the User Preferences as root. As it is more like an independent section for browsing and managing data rather than user options or settings.

```suggestion
```

itisAliRH replied:
I included this because the user preferences page displays the user's storage usage and limit and directs them to the storage dashboard and I thought it's part of this section.

jmchilton replied:
This is really nice - I like the breadcrumbs overall. I have to say my initial gut impression was that rooting the storage dashboard in user preferences made no sense to me though - my eye immediately caught on that. I had the same thought as @davelopez before I even saw his comment.

itisAliRH replied:
Thank you @davelopez and @jmchilton for your feedback. I removed them.

itisAliRH replied:
I updated the screenshots.

itisAliRH committed:
Updates breadcrumb items for storage components

Removes "User Preferences" breadcrumb item from StorageManager, StorageDashboard, and OverviewPage components as they are not tied to the user preferences

davelopez reviewed (approved):
Really cool thank you @itisAliRH!

Now the navigation is more consistent and informative of where you are :+1:
